### PR TITLE
Fix music syncing for Kodi 18 retaining Kodi 17 compatibility

### LINKED ISSUE #12: Can't sync audio
Whenever I try to add my Audio-library to kodi I get an error message:

> There was an error, you can restart kodi tp progress the scanning (message is in german)

The kodi.log shows:
```
2019-03-02 19:33:25.692 T:140312462739200  NOTICE: JELLYFIN.entrypoint.default -> path: ?mode=addlibs params: {
                                                "mode": "addlibs"
                                            }
2019-03-02 19:33:25.692 T:140312462739200  NOTICE: JELLYFIN.default -> ---<[ default ]
2019-03-02 19:33:28.714 T:140312113293056  NOTICE: JELLYFIN.full_sync -> -->[ fullsync ]
2019-03-02 19:33:28.741 T:140312113293056  NOTICE: JELLYFIN.full_sync -> starting sync with [u'16795e3accb36058bd2737dc153fc696', u'Boxsets:', u'Boxsets:Refresh']
2019-03-02 19:33:28.795 T:140312113293056  NOTICE: JELLYFIN.helper.wrapper -> Processing Hörbücher: 16795e3accb36058bd2737dc153fc696
2019-03-02 19:33:29.011 T:140312113293056  NOTICE: JELLYFIN.objects.music -> UPDATE artist [2] (T)raumschiff Surprise Periode: abfcd8d0e584252b4889713f35cef20b
2019-03-02 19:33:29.210 T:140312113293056  NOTICE: JELLYFIN.objects.music -> UPDATE album [1] Loom - Das Hörspiel: 8fd29d90d88aec910face816ac5d8b7b
2019-03-02 19:33:29.239 T:140312113293056  NOTICE: JELLYFIN.database -> ERROR:: type: <class 'sqlite3.OperationalError'> value: no such column: strArtists
2019-03-02 19:33:29.240 T:140312113293056  NOTICE: JELLYFIN.database -> [jellyfin] 39 rows updated.
2019-03-02 19:33:29.258 T:140312113293056  NOTICE: JELLYFIN.database -> ERROR:: type: <class 'sqlite3.OperationalError'> value: no such column: strArtists
2019-03-02 19:33:29.259 T:140312113293056  NOTICE: JELLYFIN.database -> [music] 184 rows updated.
2019-03-02 19:34:17.796 T:140312113293056  NOTICE: JELLYFIN.full_sync -> ERROR:: full sync exited unexpectedly
2019-03-02 19:34:17.804 T:140312113293056  NOTICE: JELLYFIN.full_sync -> --<[ fullsync ]
2019-03-02 19:34:17.805 T:140312113293056  NOTICE: JELLYFIN.library -> ERROR:: no such column: strArtists
                                            Traceback (most recent call last):
                                              File "/home/uatschitchun/.kodi/addons/plugin.video.jellyfin/resources/lib/library.py", line 524, in add_library
                                                sync.libraries(library_id, update)
                                              File "/home/uatschitchun/.kodi/addons/plugin.video.jellyfin/resources/lib/full_sync.py", line 103, in libraries
                                                self.start()
                                              File "/home/uatschitchun/.kodi/addons/plugin.video.jellyfin/resources/lib/full_sync.py", line 193, in start
                                                self.process_library(library)
                                              File "/home/uatschitchun/.kodi/addons/plugin.video.jellyfin/resources/lib/full_sync.py", line 242, in process_library
                                                media[library['CollectionType']](library)
                                              File "/home/uatschitchun/.kodi/addons/plugin.video.jellyfin/resources/lib/helper/wrapper.py", line 38, in wrapper
                                                result = func(self, dialog=dialog, *args, **kwargs)
                                              File "/home/uatschitchun/.kodi/addons/plugin.video.jellyfin/resources/lib/full_sync.py", line 418, in music
                                                obj.album(album)
                                              File "/home/uatschitchun/.kodi/addons/plugin.video.jellyfin/resources/lib/helper/wrapper.py", line 97, in wrapper
                                                return func(*args, **kwargs)
                                              File "/home/uatschitchun/.kodi/addons/plugin.video.jellyfin/resources/lib/helper/wrapper.py", line 110, in wrapper
                                                return func(self, item, e_item=e_item, *args, **kwargs)
                                              File "/home/uatschitchun/.kodi/userdata/addon_data/plugin.video.jellyfin/jellyfin/objects/music.py", line 165, in album
                                                self.update_album(*values(obj, QU.update_album_obj))
                                              File "/home/uatschitchun/.kodi/userdata/addon_data/plugin.video.jellyfin/jellyfin/objects/kodi/music.py", line 154, in update_album
                                                self.cursor.execute(QU.update_album, args)
                                            OperationalError: no such column: strArtists
2019-03-02 19:34:19.294 T:140312487917312  NOTICE: JELLYFIN.entrypoint.default -> path: ?mode=managelibs params: {
                                                "mode": "managelibs"
                                            }
2019-03-02 19:34:19.316 T:140312487917312  NOTICE: JELLYFIN.default -> ---<[ default ]
```

What to do?

### LINKED ISSUE #31: Error during Jellyfin-Kodi sync
1.JELLYFIN SERVER STATUS (OPENED)
2.JELLYFIN SYNC QUEUE PLUGIN STATUS (INSTALLED)

SCREENSHOOT IN THE ATTACH FILE

image

### PR DESCRIPTION
**Pull request by jmeacoe**
Fixes music syncing for kodi 18 while still usable by kodi 17.
Also removes the message in kodi 18 to scan media.
tested in kodi 17 and 18.

Fixes #12 
Fixes #31

### CODE REVIEW

**Comment by jmeacoe** [comment]
This pull request also fixes #31 which is the strArtists issue in the music72 db.

File: resources/lib/objects/kodi/music.py [hunk 1]
```diff
@@ -146,58 +150,77 @@ def get_album(self, album_id, name, musicbrainz, artists=None, *args):
...
+        else:
+            self.cursor.execute(QU.update_song72, args)
 
     def link_song_artist(self, *args):
         self.cursor.execute(QU.update_song_artist, args)
 
     def link_song_album(self, *args):
-        self.cursor.execute(QU.update_song_album, args)
+        if self.version_id < 72:
+            self.cursor.execute(QU.update_song_album, args)
```
Comment (oddstr13) [inline resources/lib/objects/kodi/music.py]: No Kodi 18 equivalent for this?

File: resources/lib/objects/kodi/music.py [hunk 2]
```diff
@@ -146,58 +150,77 @@ def get_album(self, album_id, name, musicbrainz, artists=None, *args):
...
+                self.cursor.execute(QU.get_album_artist, (album_id,))
+            else:
+                self.cursor.execute(QU.get_album_artist72, (album_id,))
             curr_artists = self.cursor.fetchone()[0]
         except TypeError:
             return
 
         if curr_artists != artists:
-            self.update_album_artist(artists, album_id)
+                self.update_album_artist(artists, album_id)
```
[the remark this answers is not in the document]
Comment (jmeacoe) [reply resources/lib/objects/kodi/music.py]: looks accidental, as I don't recall changing anything in that section.

File: resources/lib/objects/kodi/music.py (hunk 1 above)
Comment (jmeacoe) [reply resources/lib/objects/kodi/music.py]: Kodi 18 removed that table completely.

**Comment by jmeacoe** [comment]
Kodi MusicDB goes from Kodi 17 - Music60 to Kodi 18 -Music72:
https://kodi.wiki/view/Databases#Database_Versions
I will fix the indentation when I get home from work.

**Comment by macbean101** [comment]
referred here from another (i guess the same?) issue.  installed the supplied updated plugin and initially it didn't work.  But after renaming my kodi config folder to start fresh, this started scanning without issue (so far).

**Comment by Gomez** [comment]
Confirmed. Current master works (https://github.com/jellyfin/jellyfin-kodi/archive/master.zip). But i had to reset my kodi too.

Is a reset really necessary?

**Comment by macbean101** [comment]
I just finalized my move from emby to jellyfin.  This time, instead of resetting completely, i just created a new profile and made it my default.  This way at least everything I needed was already downloaded and installed.

**Comment by Renaud11232** [comment]
Looks nice, any idea on when it will be released on the plugin repository ?

**Comment by oddstr13** [comment]
I want to improve debug loging first, to make new issues easier to track down

**Pull request merged**